webxrlayers-1.bs

```diff
@@ -769,21 +771,26 @@ is initialized.
 <pre class="idl">
 dictionary XRProjectionLayerInit {
   XRTextureType textureType = "texture";
+  GLenum colorFormat = 0x1907; // RGBA
+  GLenum depthFormat = 0x84FA; // UNSIGNED_INT_24_8_WEBGL
   boolean depth = true;
   boolean stencil = false;
-  boolean alpha = true;
   double scaleFactor = 1.0;
 };
 </pre>
 
 The <dfn dict-member for="XRProjectionLayerInit">textureType</dfn> attribute defines the type of texture that the layer will have.
 
+The <dfn dict-member for="XRProjectionLayerInit">colorFormat</dfn> attribute defines the data type of the color texture data.
+The default value is {{RGBA}}.
+
+The <dfn dict-member for="XRProjectionLayerInit">depthFormat</dfn> attribute defines the data type of the depth texture data.
+The default value is {{UNSIGNED_INT_24_8_WEBGL}}.
```

Comment:
Hm... There's a couple of issues with the default enum here.

For one, this isn't a base WebGL 1 enum, but that makes me realize that WebGL 1 doesn't actually support depth textures without an extension. Yuck. 😫 I'm going to file a separate issue on that, because we'll have to address it no matter what approach we take here.

Second, ignoring the first issue, this changes the defaults a bit in that it implies stencil is present, which the original booleans default to false. Granted, until you hit WebGL2 there's not a way to require 24 bit depth without stencil, and the precision loss doesn't make for a good default.

Kinda wondering if the right approach is to actually default to `gl.NONE`, let WebGL 1 baseline users use their own renderbuffers, and force devs to make an explicit choice about which format they use otherwise. It'll be a bit of a departure from the patterns we currently have set up, but if anything now would be the time to make those kind of breaking changes. (I'd be happy to reflect any default patterns we establish here in the WebGPU layers for consistency.)

Reply:
> For one, this isn't a base WebGL 1 enum, but that makes me realize that WebGL 1 doesn't actually support depth textures without an extension. Yuck. 😫 I'm going to file a separate issue on that, because we'll have to address it no matter what approach we take here.

The spec already deals with that. For instance see step 4  in [this](https://immersive-web.github.io/layers/#allocate-depth-textures) algorithm.

> Second, ignoring the first issue, this changes the defaults a bit in that it implies stencil is present, which the original booleans default to false. Granted, until you hit WebGL2 there's not a way to require 24 bit depth without stencil, and the precision loss doesn't make for a good default.
> 
> Kinda wondering if the right approach is to actually default to `gl.NONE`, let WebGL 1 baseline users use their own renderbuffers, and force devs to make an explicit choice about which format they use otherwise. It'll be a bit of a departure from the patterns we currently have set up, but if anything now would be the time to make those kind of breaking changes. (I'd be happy to reflect any default patterns we establish here in the WebGPU layers for consistency.)

Maybe we can get rid of the depth and stencil booleans in the init structure and then rely on the enum to determine if depth or stencil are requested.

Reply:
> The spec already deals with that. For instance see step 4 in this algorithm.

Awesome! Didn't see that earlier (and it mostly makes #223 moot.) I do think we should re-evaluate the defaults, since the current setup would cause it to always fail on baseline WebGL 1 contexts.

> Maybe we can get rid of the depth and stencil booleans in the init structure and then rely on the enum to determine if depth or stencil are requested.

Yeah, I think that feels best.